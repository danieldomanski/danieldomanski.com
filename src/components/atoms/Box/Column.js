import styled from 'styled-components'
import { space, color, layout, flexbox } from 'styled-system'

const Column = styled.div(
  {
    minWidth: 0,
    display: 'flex',
    flexDirection: 'column',
  },
  space,
  color,
  layout,
  flexbox
)

export default Column
