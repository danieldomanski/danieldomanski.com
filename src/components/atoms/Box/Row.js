import styled from 'styled-components'
import { space, color, layout, flexbox } from 'styled-system'

const Row = styled.div(
  {
    minWidth: 0,
    display: 'flex',
  },
  space,
  color,
  layout,
  flexbox
)

export default Row
