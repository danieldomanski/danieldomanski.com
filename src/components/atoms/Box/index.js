import styled from 'styled-components'
import {
  space,
  typography,
  color,
  layout,
  flexbox,
  border,
  shadow,
  position,
} from 'styled-system'

const Box = styled.div(
  {
    minWidth: 0,
  },
  space,
  color,
  border,
  layout,
  flexbox,
  shadow,
  typography,
  position
)

export default Box
