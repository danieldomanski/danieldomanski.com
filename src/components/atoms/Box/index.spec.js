import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import Box from './index'

afterEach(cleanup)

test('<Box> renders with different styled-system category props', () => {
  const { container } = render(
    <Box display="flex" width={200} bg="#fff">
      <div>Test.</div>
    </Box>
  )

  expect(container.firstChild).toMatchSnapshot()
})
