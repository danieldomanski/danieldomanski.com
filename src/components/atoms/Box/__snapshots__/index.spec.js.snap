// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box> renders with different styled-system category props 1`] = `
.c0 {
  min-width: 0;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 200px;
}

<div
  class="c0"
  display="flex"
  width="200"
>
  <div>
    Test.
  </div>
</div>
`;